include/input_handler.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include <glad/glad.h>
+#include <GLFW/glfw3.h>
+#include <window.h>
+#include <functional>
+
+namespace arch{
+    class InputHandler{
+    public:
+        InputHandler(const InputHandler&) = delete;
+        InputHandler(const InputHandler&&) = delete;
+        static InputHandler& get();
+        void set_current_window(GLFWwindow* window);       
+
+    private:
+        InputHandler();
+        static InputHandler input_handler_instance;
+        static GLFWwindow *_window;
```

Comment:
I don't think we need to bind Close Window, Maximize Window etc. in Input Manager. I think we can get rid of these functions if it means removing this static pointer property. It is always good for systems to be as much decoupled as possible.